### ISSUE TITLE
[backwards compatbility issue] GraphQL::Query::Result.to_h has GraphQL named classes.

### ISSUE THREAD

**Issue by danlo** [rmosolgo/graphql-ruby#3516]
**Describe the bug**

ruby Marshal(ing) the result of a GraphQL::Query::Result.to_h contains GraphQL named classes, which prevents un-marshalling on destination machines that do not have graphql installed.

**Versions**

`graphql` version: broke in 1.12.11 (works in 1.12.10)

**Steps to reproduce**

Use version 1.12.11 or 1.12.12

Execute any graphql request
```ruby
result = Sample.execute(query)
(byebug) result
#<GraphQL::Query::Result @query=... @to_h={"data"=>{"xxx"=>true}}>
(byebug) result.to_h
{"data"=>{"xxx"=>true}}
(byebug) result.to_h['data']
{"xxx"=>true}
(byebug) result.to_h['data'].class
GraphQL::Execution::Interpreter::Runtime::GraphQLResultHash
```

**Expected behavior**

result.to_h should return a complete hash containing only ruby Hash(es) and Array(s)

In version 1.12.10
```ruby
(byebug) pp result['data'].class
Hash
```

**Actual behavior**

result.to_h returns ['data'] which is of class ```GraphQL::Execution::Interpreter::Runtime::GraphQLResultHash``` 

when this is "Marshal.dump()" the class name is saved and the destination is expected to have graphql installed in order to unpack the marshal'd data.

---

Thank you!

**Comment by rmosolgo** [comment]
Sorry for the problem, and thanks for the careful writeup of the issue! I'll investigate some way to put back the old behavior of `.to_h`.

**Comment by rmosolgo** [comment]
This should be fixed by #3533 in 1.12.13 which I'll release later today :+1:
